src/components/resizable.tsx

```diff
@@ -0,0 +1,164 @@
+import '@isrd-isi-edu/chaise/src/assets/scss/_resizable.scss';
+import { useRef, useEffect, useState } from 'react';
+import { fireCustomEvent } from '@isrd-isi-edu/chaise/src/utils/ui-utils';
+
+/* 
+Component Usage:
+    This component is used to implement a horizontal dragable and resizable component/div
+
+Props:
+    left - the left-pane component in the resizable layout
+    right - the right-pane component in the resizable layout
+    className - to apply custom styles on the component
+    minWidth - the minimum width limit on the left-pane component | defaults to 250px
+    maxWidth - the maximum width limit on the left-pane component | defaults to 450px
+    initialWidth - the initial width for the left-pane component | defaults to 270px
+    convertMaxWidth - use this flag in case you pass maxWidth in vw units instead of px units
+
+Note:
+    To resize other components based on the resized value left-pane component add an event listener and listen to the 
+    'resizable-width-change' event.
+
+    For usage refer to the recordset.tsx component
+*/
+
+
+const vwToPx = (value: number) => {
+    const e = document.documentElement;
+    const g = document.getElementsByTagName('body')[0];
+    const x = window.innerWidth || e.clientWidth || g.clientWidth;
+
+    const result = (x * value) / 100;
+    return result;
+}
+
+type LeftPaneProps = {
+    children: (ref: React.RefObject<HTMLDivElement>) => JSX.Element,
+    leftWidth: number | undefined,
+    setLeftWidth: (value: number) => void
+};
+
+const LeftPane = ({ children, leftWidth, setLeftWidth }: LeftPaneProps): JSX.Element => {
+    const leftRef = useRef<HTMLDivElement>(null);
+
+    useEffect(() => {
+        if (leftRef?.current) {
+            if (!leftWidth) {
+                setLeftWidth(leftRef.current.clientWidth);
```

Comment:
Is there any way that we can avoid passing this setter function? Can't we assume `leftWidth` will always have a default value?

Reply:
done